core/common/src/main/java/alluxio/util/UnderFileSystemUtils.java

```diff
@@ -67,5 +69,22 @@ public static void touch(final String path) throws IOException {
     os.close();
   }
 
+  /**
+   * Deletes the specified path from the specified under file system if it exists. This will not
+   * delete nonempty directories.
+   *
+   * @param ufs the under file system to delete from
+   * @param path the path to delete
+   */
+  public static void deleteIfExists(UnderFileSystem ufs, String path) {
+    try {
+      if (ufs.exists(path)) {
+        ufs.delete(path, false);
+      }
+    } catch (IOException e) {
+      throw Throwables.propagate(e);
```

Comment:
The javadoc for UnderFileSystem just says `@throws IOException if a non-Alluxio error occurs` for both `exists` and `delete`. This method is never expected to throw an exception.

I don't think the lack of atomicity is a problem. The contract is that if the file path exists, it gets deleted. If it gets deleted between the exists and the delete, the delete will return false and the result is the same (no more file)